nativescript-3.2.0-memory-management/README.md

```diff
@@ -0,0 +1,116 @@
+# NativeScript 3.2.0: Memory Management
+NativeScript binds very small JavaScript instances to the native instances. These let developers access and consume the native APIs from JavaScript. Here is explained the life cycle of JavaScript and native instance.
+
+ - [Terms](#terms)
+ - [iOS](#ios)
+ - [Android](#android)
+ - [Call to Arms](#call-to-arms)
+
+## Terms
+*Disclaimer: Please, lookup these in a proper source. I made them myself.*
+
+**native instance** - Objective-C class instance or Java class instance based on the context.
+
+**reference counting** - iOS's Objective-C runtime uses reference counting. The instances have internal counters that can be incremented and decremented. Each time a strong reference is set to point to an instance, the instance have its reference count incremented. Each time a strong reference change, the previous instance it pointed to, have its reference count decremented. When decremented to 0, the instance is deallocated.
+
+**GC** - garbage collection in general. When GCs run, they first block the threads to collect all root instances from the stack. Then resume the execution until they mark all reachable objects in a separate thread. Then block again the threads to complete the marking. And at the end start finalizing and deallocating unreachable instances. While the actual GC implementation may be much more sophisticated, all implementations in virtual machines used for UI aim at minimizing the time the main thread is blocked. Java, iOS's JavaScriptCore and Android's V8 are 3 state of the art virtual machines used in NativeScript, that have garbage collectors.
+
+**set weak/strong** - when an instance is set strong it is ensured to survive a garbage collection along with all instances reachable from it. When an instance is set weak, it can be collected if it is no longer reachable from other live instances.
+
+**splice** - let's introduce a new NativeScript term. Splice we will call a bond, made between a native instance and its representation in JavaScript by a JavaScript instance. It is possible that the splice is instantiated first in native in the cases of the iOS AppDelegate class, Android's Application, Activity and Fragment classes.
```

Comment:
`Splice we will call a bond...`
Splice: a bond ...    |   We will call "Splice" a bond ...